src/FastPriorityQueue.php

```diff
@@ -0,0 +1,300 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Stdlib;
+
+use Iterator;
+use Countable;
+use Serializable;
+
+/**
+ * This is an efficient implementation of an integer priority queue in PHP
+ *
+ * This class acts like a queue with insert() and extract(), removing the
+ * elements from the queue and it also acts like an Iterator without removing
+ * the elements. This behaviour can be used in mixed scenarios with high
+ * performance boost.
+ */
+class FastPriorityQueue implements Iterator, Countable, Serializable
+{
+    const EXTR_DATA     = 0x00000001;
+    const EXTR_PRIORITY = 0x00000002;
+    const EXTR_BOTH     = 0x00000003;
```

Comment:
Makes sense; open a pull request for the change, please.
On Sep 18, 2015 7:39 AM, "Maks3w" <notifications@github.com> wrote:

> In src/FastPriorityQueue.php
> <https://github.com/zendframework/zend-stdlib/pull/19#discussion_r39850085>
> :
>
> > +use Countable;
> > +use Serializable;
> > +
> > +/**
> > + * This is an efficient implementation of an integer priority queue in PHP
> > + *
> > + * This class acts like a queue with insert() and extract(), removing the
> > + * elements from the queue and it also acts like an Iterator without removing
> > + * the elements. This behaviour can be used in mixed scenarios with high
> > + * performance boost.
> > + */
> > +class FastPriorityQueue implements Iterator, Countable, Serializable
> > +{
> > +    const EXTR_DATA     = 0x00000001;
> > +    const EXTR_PRIORITY = 0x00000002;
> > +    const EXTR_BOTH     = 0x00000003;
>
> Why not use the constant then?
>
> const EXTR_DATA = SplPriorityQueue::EXTR_DATA;
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendframework/zend-stdlib/pull/19/files#r39850085>.
>